src/client/app/containers/CompareChartContainer.js

```diff
@@ -58,56 +60,60 @@ function mapStateToProps(state, ownProps) {
 		readingsData = state.readings.bar.byMeterID[ownProps.id][timeInterval][barDuration];
 	}
 	if (readingsData !== undefined && !readingsData.isFetching) {
-		// Sunday needs special logic
-		if (soFar !== 0) {
-			// Calculate currentWeek
-			for (let i = 1; i <= soFar; i++) {
-				currentWeek += readingsData.readings[readingsData.readings.length - i][1];
-			}
-			// Calculate lastWeek
-			for (let i = 0; i < 7; i++) {
-				lastWeek += readingsData.readings[readingsData.readings.length - (8 + i) - soFar][1];
-			}
+		if (readingsData.readings.length < 7) {
+			soFar = moment().hour();
+		} else if (readingsData.readings.length < 14) {
+			soFar = moment().diff(moment().startOf('week'), 'days');
+		} else {
+			soFar = moment().diff(moment().startOf('week').subtract(21, 'days'), 'days');
+		}
 
-			// Calculate currentLastWeek
-			for (let i = 1; i <= soFar; i++) {
-				currentLastWeek += readingsData.readings[readingsData.readings.length - i - 7][1];
-			}
+		// Calculates current
+		for (let i = readingsData.readings.length - soFar; i < readingsData.readings.length; i++) {
+			current += readingsData.readings[i][1];
+		}
+		// Calculate prev
+		for (let i = 0; i < readingsData.readings.length - soFar; i++) {
+			prev += readingsData.readings[i][1];
+		}
+		// Calculate currentPrev. Have to special case Sunday. Fixing this is a good first issue since we now have hour data.
+		if (soFar === 0) {
+			currentPrev = Math.round((readingsData.readings[0][1] / 24) * moment().hour());
```

Comment:
Recommend not commenting that this is a good first issue, but instead opening a GitHub issue.